create visualization graph

YukaUU opened the pull request:
<!--- Provide a short summary of your changes in the Title -->
Add visualization panel to viewer page.
## Summary
<!--- Describe the changes in more detail. Feel free to include screenshots if relevant. -->
Click the icon to the left of the toolbar's tutorial to open the side menu on the right side.
There you can see a graph visualizing the annotation information
It shows the relationship between human:annotation and zooming.
![all](https://github.com/user-attachments/assets/7eeb3ce6-60a6-4b41-986e-9d87b47a9e13)
## Motivation
<!--- Why have you made this Pull Request? Does this fix an open issue? -->
To visualize pathologist annotation information
## Testing
<!--- Has this been tested beyond the automatic hooks? How so? -->
The graph and the results with collection:Mark.
Annotation:0
![d0](https://github.com/user-attachments/assets/7f102cd3-e2b2-4e52-8ed8-e55aa7451230)
![d0_log](https://github.com/user-attachments/assets/31258ee7-dbf4-4dcd-934e-509fd76e4634)
Annotation:1
![d1](https://github.com/user-attachments/assets/0955a632-77ab-4ed4-8730-4357b07db1ee)
![d1_log](https://github.com/user-attachments/assets/8353d767-dda2-4f52-8d0a-7a07ee3a7aed)
Annotation:9
![d9](https://github.com/user-attachments/assets/67df5e5e-243a-47c3-ab3d-9c0a61b4c790)
![d9_log](https://github.com/user-attachments/assets/13935f40-8356-4c44-9010-952240c3eb32)

birm reviewed (changes requested):
Can we call this new field something a little more reasable, like viewerState?

Maybe you want to save it somewhere inside, so that each drawn object has its own x,y,zoom, even if they're grouped together? I noted some issues when testing this in https://github.com/camicroscope/caMicroscope/issues/1013

YukaUU commented:
> Can we call this new field something a little more reasable, like viewerState?

OK! 
 I change new field name.


> Maybe you want to save it somewhere inside, so that each drawn object has its own x,y,zoom, even if they're grouped together? I noted some issues when testing this in #1013

That's right.
I will fix it.

YukaUU commented:
@birm 
I changed new field name from states to viewerStates.https://github.com/camicroscope/caMicroscope/pull/1011#:~:text=will%20fix%20it.-,change%20name%20from%20states%20to%20viewerStates,-10a8801

Test

1. Show x, y and zooming in pop panel
![image](https://github.com/user-attachments/assets/3441c1f4-bf5f-4641-96b6-28f2194c74b0)

2.log (send data to DB)
![change_states_viewerStates](https://github.com/user-attachments/assets/5f00342a-bb58-4167-97a3-b7184b6ceefa)

3.log:DB collection:Mark
![chabge_states_viewerStates](https://github.com/user-attachments/assets/f4d38f79-d512-4b48-bd3c-62dbb54080fd)

YukaUU commented:
@birm

> Maybe you want to save it somewhere inside, so that each drawn object has its own x,y,zoom, even if they're grouped together? I noted some issues when testing this in https://github.com/camicroscope/caMicroscope/issues/1013

Not all annotations saved have (accurate) info saved #1013https://github.com/camicroscope/caMicroscope/issues/1013#issue-2414149542

I  fixed it!

**Test**
1. Annotation: Multi mode
<img width="1438" alt="Annotation: Multi mode " src="https://github.com/user-attachments/assets/1ac7a6ec-5d0e-48ef-96e3-d8ff6fa52f96">

2. log (send data to DB)
![image](https://github.com/user-attachments/assets/3a854fc6-a43a-4a2b-9766-d04184b6fd32)

3.log:DB collection:Mark
![image](https://github.com/user-attachments/assets/26a6b443-0d4d-4223-ace1-6d39e695a235)

4. Check pop up panel
![image](https://github.com/user-attachments/assets/8d978d4f-fb9c-4e85-b405-1acd4afc2faf)
![image](https://github.com/user-attachments/assets/92a16ce7-73a1-4a7a-a748-b90986bd1dd8)
![image](https://github.com/user-attachments/assets/358b08c2-7edd-4af5-9353-13926f76d817)
![image](https://github.com/user-attachments/assets/88e74335-a8e2-4e5a-9d36-98ea63aa82db)
![image](https://github.com/user-attachments/assets/4d2370c8-c5b5-4cdf-bd18-46deae97415c)

YukaUU commented:
Delete viewerStates 
Past PR did not support Multi Mode.　https://github.com/camicroscope/caMicroscope/pull/966
I fixed it https://github.com/camicroscope/caMicroscope/pull/1011#issuecomment-2236869813
This commit was also deleted.https://github.com/camicroscope/caMicroscope/pull/1011#issuecomment-2236230522

**Test**
1. Annotation: Single mode
![image](https://github.com/user-attachments/assets/fa336fdf-efc2-433d-87c7-48ac1ef5ef28)
log:DB collection:Mark
![image](https://github.com/user-attachments/assets/83b7d85a-0da5-4b12-89f4-b215c27874ef)

2.Annotation: Multi mode
![image](https://github.com/user-attachments/assets/74e67089-9a1b-478c-9779-124e24231c90)
log:DB collection:Mark
![image](https://github.com/user-attachments/assets/4f5e01b8-a155-42e7-b0ce-94460751fba8)

birm commented on core/extension/openseadragon-canvas-draw-overlay.js:
```diff
@@ -1023,6 +1024,11 @@
+          viewerStates: {
+            x: StatesHelper.getCurrentStates(true).x,
+            y: StatesHelper.getCurrentStates(true).y,
+            z: StatesHelper.getCurrentStates(true).y
```
```suggestion
            z: StatesHelper.getCurrentStates(true).z
```
I'm pretty sure this is why this is happening:
![Screenshot 2024-07-19 at 10 49 08 AM](https://github.com/user-attachments/assets/0267d02e-9080-4385-8a77-80e170a9dc07)
![Screenshot 2024-07-19 at 10 49 02 AM](https://github.com/user-attachments/assets/c363255a-51bb-4812-830b-aaa2f1335f6e)

YukaUU replied:
I also noticed a mistake in my local and was fixing the same area.
Thank you!!

birm reviewed (changes requested):
Very close; just one hopefully quick thing I noticed!

YukaUU commented:
@birm 
Refactored to remove unnecessary code, etc
I resolved all comments  and issue.
（https://github.com/camicroscope/caMicroscope/pull/1011#pullrequestreview-2183556879 , https://github.com/camicroscope/caMicroscope/issues/1013)
Please check it
### Test
Visualization Graph
![image](https://github.com/user-attachments/assets/c002a889-2d89-400b-b659-b7cbcac80d9c)
log (send data to DB)
![image](https://github.com/user-attachments/assets/1583239a-d8b5-4277-ad4b-9d964fd4f3ef)
log:DB collection:Mark
![image](https://github.com/user-attachments/assets/1be070e7-e54c-4277-af2e-06635f8869cc)
Pop up panel
![image](https://github.com/user-attachments/assets/c829e29f-9664-4181-ad1b-c36d86acd1a4)
![image](https://github.com/user-attachments/assets/d28408ab-9c28-404d-a19c-2e4ad150320f)
![image](https://github.com/user-attachments/assets/be4a93c9-cc6e-4219-bc34-c15f5ecea648)
![image](https://github.com/user-attachments/assets/d5dc7252-42d0-408e-a3ee-89acdcf16020)
![image](https://github.com/user-attachments/assets/b633d36c-f3ae-406c-814a-4dc53d0a8896)

birm reviewed (approved):
Great work, and good fixes!
